src/test/coll.rs

```diff
@@ -545,30 +502,26 @@ async fn find_allow_disk_use_not_specified() {
     let coll = event_client
         .database(function_name!())
         .collection(function_name!());
+    coll.find(None, options).await.unwrap();
 
-    let find_opts = FindOptions::builder().build();
-    coll.find(None, find_opts).await.unwrap();
-
-    assert!(event_client
-        .command_events
-        .read()
-        .unwrap()
-        .iter()
-        .all(|event| match event {
-            CommandEvent::CommandStartedEvent(CommandStartedEvent {
-                command_name,
-                command,
-                ..
-            }) => {
-                if command_name == "find" {
-                    match command.get_bool("allowDiskUse") {
-                        Ok(_) => false,
-                        _ => true,
-                    }
-                } else {
-                    true
-                }
+    let events = event_client.command_events.read().unwrap();
+    let mut iter = events.iter().filter(|event| match event {
+        CommandEvent::CommandStartedEvent(CommandStartedEvent { command_name, .. }) => {
+            command_name == "find"
+        }
+        _ => false,
+    });
+
+    let event = iter.next().unwrap();
+    let allow_disk_use = match event {
+        CommandEvent::CommandStartedEvent(CommandStartedEvent { command, .. }) => {
+            match command.get_bool("allowDiskUse") {
```

Comment:
Another way to do this would be to use the method [`Result::ok`](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok), which converts `Ok(value)` to `Some(value)` and `Err(..)` to `None`